annotation/tests/test_tasks_services.py

```diff
@@ -1,30 +1,240 @@
 import re
 from copy import deepcopy
-from typing import Dict, List, Optional, Tuple
-from unittest.mock import MagicMock, Mock, call, patch
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Tuple
+from unittest.mock import ANY, MagicMock, Mock, call, patch
 from uuid import uuid4
 
 import pytest
 from fastapi import HTTPException
+from tenant_dependency import TenantData
 
-from annotation.errors import FieldConstraintError
+import annotation.tasks.services as services
+from annotation.errors import CheckFieldError, FieldConstraintError
 from annotation.filters import TaskFilter
 from annotation.jobs.services import ValidationSchema
-from annotation.models import File, ManualAnnotationTask
-from annotation.schemas.tasks import ManualAnnotationTaskInSchema
-from annotation.tasks.services import (
-    check_cross_annotating_pages,
-    create_annotation_task,
-    filter_tasks_db,
-    read_annotation_task,
-    read_annotation_tasks,
-    remove_additional_filters,
-    validate_files_info,
-    validate_ids_and_names,
-    validate_task_info,
-    validate_user_actions,
-    validate_users_info,
+from annotation.models import (
+    AnnotatedDoc,
+    AnnotationStatistics,
+    File,
+    ManualAnnotationTask,
 )
+from annotation.schemas.annotations import PageSchema, ParticularRevisionSchema
+from annotation.schemas.tasks import (
+    AgreementScoreComparingResult,
+    AgreementScoreServiceResponse,
+    AnnotationStatisticsInputSchema,
+    ManualAnnotationTaskInSchema,
+    ResponseScore,
+    TaskMetric,
+    TaskStatusEnumSchema,
+)
+
+
+@pytest.fixture
+def mock_task_revisions():
+    mock_revision = MagicMock(spec=AnnotatedDoc)
+    mock_revision.pages = {"1": ["data1"], "2": ["data2"]}
+    mock_revision.failed_validation_pages = [1, 2]
+    mock_revision.validated = [1, 2]
+    yield mock_revision
+
+
+@pytest.fixture
+def mock_metric():
+    metric = MagicMock()
```

Comment:
any reason using mock here? why not the real model?

Reply:
Ok:)